### PR TITLE
DDF-3549 Download links for federated products in Intrigue should point to the local DDF instead of the remote one

### PR DESCRIPTION
**Pull request by Lambeaux**
Abbreviated review in effect. 

#### What does this PR do?
Forward port of https://github.com/codice/ddf/pull/2844

#### Who is reviewing it? 
@brjeter 
@blen-desta 

#### Choose 2 committers to review/merge the PR. 
@brendan-hofmann
@coyotesqrl 

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Lambeaux** [comment]
Build now.

File: catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java [hunk 1]
```diff
@@ -68,10 +70,10 @@
 
   private Attribute idAttribute;
 
-  // Suppressing Warnings and using ArrayList rather than List here because
-  // ArrayList implements Serializable (List does not).
+  // Suppressing Warnings and using HashSet rather than Set here because
+  // HashSet implements Serializable (Set does not).
   @SuppressWarnings("all")
```
Comment (brendan-hofmann) [inline catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java]: ✏️ This suppression could be more specific. Something about interface vs implementation declarations iirc.

File: catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java [hunk 2]
```diff
@@ -160,7 +162,7 @@ protected void setMetacardType(MetacardType metacardType) {
     @XmlElement(name = "string", namespace = METACARD_URI, type = StringElement.class),
     @XmlElement(name = "stringxml", namespace = METACARD_URI, type = StringxmlElement.class)
   })
-  protected List<Attribute> getAttributes() {
+  protected Set<Attribute> getAttributes() {
     return attributes;
```
Comment (brendan-hofmann) [inline catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java]: ❓ Should this be a defensive copy?

Comment (Lambeaux) [reply catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java]: Can't be. Serialization depends upon having access to the underlying structure. Making a defensive copy breaks a lot of unit tests and work require a deal of rework if it was even possible in the first place.

File: catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java (hunk 1 above)
Comment (Lambeaux) [reply catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java]: Sure, but given this is a straight forward port I'm going to leave it be for now.

**Comment by Lambeaux** [comment]
Build now.

**Pull request merged**